ipfs/ipfs-docs
# testing for https://github.com/ipfs/ipfs-docs/pull/1330

🥺  DO NOT MERGE THIS BRANCH - it's for testing purposes only.

The purpose of this branch is to provide a horribly hacky way to test the GH action implemented in https://github.com/ipfs/ipfs-docs/pull/1330 before said PR is actually merged into the `main` branch of `ipfs\ipfs-docs `.

**How to use this branch:**
1. **PLEAAAAAASE Don't merge this** - I am going to delete this branch without merging after we run all of the tests that we need to run.
2. Pull this PR down locally. Instructions are at https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally?tool=cli#modifying-an-active-pull-request-locally. LMK if you have issues getting the PR.
3. Pick a test case (listed below).
4. Comment on this PR with the test case that you're picking up.
5. Locally, do the markdown mods that you need to do to complete the test case. _One test case per commit please_ - this will make it easier to track what all has been tested.
6. Git add the changed file
7. Git commit with a message that describes:
    - Test case
    - file changed / location, 
    - Expected outcome
    separated by commas. Example commit message: "Github link test, README.md line 133, should fail"
8. Git push 
9. Navigate back to this PR in the browser
10. Wait for the CI/CD check to complete.
11. Edit your comment with the outcome of the test. If the check doesn't do what it's supposed to, I will investigate and fix it. If it does, great! Thanks for the help.

**Tests needed**
1. Github links
2. pl domain links
3. Weirdly formatted links
4. Links in the front matter 
5. Headers of varying depths
6. Internal links that are not relrefs (per Johnny's comment https://github.com/ipfs/ipfs-docs/pull/1330#issuecomment-1322279186 in the future, internal links wont be an issue since we will only be using relrefs. However, with the current ipfs-docs, we probably have some internal links that are standard MD links, so this tool applies here)
